src/html-purify.js

```diff
@@ -52,23 +52,26 @@ See the accompanying LICENSE file for terms.
 	    break;
 
         case derivedState.TransitionName.FROM_TAG_ATTR_TO_DATA:
-            if (prevState === 35 ||
-                prevState === 36 || 
+            if (prevState === 36 || 
                 prevState === 40) {
                 this.attrVals[attributeName] = attributeValue;
             }
+            // boolean attributes may not have a value
+            if (prevState === 35) {
```

Comment:
FYI - I am also adding a list of boolean attributes. We will set new value of non-boolean attribute with no value to ""